commit ca112fb6dec2157477c3c952bcfddcdc0aece093
Author:     Isaac Yang <isaacy@nvidia.com>
AuthorDate: Fri Jun 2 09:43:08 2023 -0700
Commit:     Isaac Yang <isaacy@nvidia.com>
CommitDate: Wed Jun 21 15:17:45 2023 -0700

    Add missing cert chain verification on submitted job signatures
    Add unit test on verify_folder_signature
---
 nvflare/private/fed/server/job_runner.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/nvflare/private/fed/server/job_runner.py b/nvflare/private/fed/server/job_runner.py
index 203720b5..3d123abc 100644
--- a/nvflare/private/fed/server/job_runner.py
+++ b/nvflare/private/fed/server/job_runner.py
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import json
+import os
 import shutil
 import threading
 import time
@@ -156,7 +157,8 @@ class JobRunner(FLComponent):
                     secure_train = kv_list.get("secure_train", True)
                     if secure_train:
                         app_path = workspace.get_app_dir(job.job_id)
-                        if not verify_folder_signature(app_path):
+                        root_ca_path = os.path.join(workspace.get_startup_kit_dir(), "rootCA.pem")
+                        if not verify_folder_signature(app_path, root_ca_path):
                             err = "job signature verification failed"
                             deploy_detail.append(f"server: {err}")
                             raise RuntimeError(f"Failed to verify app '{app_name}': {err}")
